content/en/docs/setup/independent/troubleshooting-kubeadm.md

```diff
@@ -215,27 +215,64 @@ kubectl -n kube-system patch --type json daemonset kube-proxy -p "$(cat <<'EOF'
 [
     {
         "op": "add",
-        "path": "/spec/template/spec/containers/0/env",
-        "value": [
-            {
-                "name": "NODE_NAME",
-                "valueFrom": {
-                    "fieldRef": {
-                        "apiVersion": "v1",
-                        "fieldPath": "spec.nodeName"
-                    }
-                }
+        "path": "/spec/template/spec/volumes/-",
+        "value": {
+                "name": "shared-data",
+                "mountPath": "/shared-data"
+            }
+    },
+    {
+        "op": "add",
+        "path": "/spec/template/spec/containers/0/volumeMounts/-",
+        "value": {
+                "name": "shared-data",
+                "mountPath": "/shared-data"
             }
-        ]
+    },
+    {
+        "op": "replace",
+        "path": "/spec/template/spec/containers/0/command/1",
+        "value": "--config=/shared-data/config.conf"
     },
     {
         "op": "add",
-        "path": "/spec/template/spec/containers/0/command/-",
-        "value": "--hostname-override=${NODE_NAME}"
+        "path": "/spec/template/spec/initContainers",
```

Comment:
Why the switch to an initContainer here? Is the `--hostname-override` command line option no longer taking precedence over the component config?

Reply:
That's exactly right. I found that `--hostname-override` did not take precedence over the component config. I tested with v1.10.4 and v1.11.1 and reported my findings in https://github.com/kubernetes/kubeadm/issues/857.

Thanks for taking a look!

Reply:
@dlipovetsky @detiber 
that's very odd. so for kube-proxy `--hostname-override` just overrides the default config:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server.go#L144

i don't anything should modify it at that point, right?

@stewart-yu 
any ideas about this?

Reply:
@neolit123 I ran experiments and found that the config overrides flags (Please see my findings in https://github.com/kubernetes/kubeadm/issues/857#issuecomment-408576625). 

Looking at the code, I find that the flags updated a config struct, but later the configuration file is unmarshalled into a new config struct, and [the pointer to the config struct is overwritten](https://github.com/kubernetes/kubernetes/blob/release-1.11/cmd/kube-proxy/app/server.go#L204), effectively discarding all the flag values. (Maybe this is by design, since the flags are [deprecated](https://github.com/kubernetes/kubernetes/blob/f07651cfa448e8c93f843449f3ce98e0ce7e15d9/cmd/kube-proxy/app/server.go#L195).)

Reply:
hm, even if the flags are deprecated they have a period of N releases where they should still work as expected. if this is by design, then we have a bit of an issue here.

my problem is that i don't know who to poke about kube-proxy.
i usually ask @stewart-yu but let's ask the whole sig-network about this...

/cc @kubernetes/sig-network-bugs 
please, have a look at the observations by @dlipovetsky above.

Reply:
The config file and flags appear to have mutually exclusive starting with [the commit that introduced support for the config file](https://github.com/kubernetes/kubernetes/commit/43cb024402e6992418b2197228889633b82ddb75#diff-cf93bbed37202b7d58b5841f07ddd89fR177). That was in v1.7.